sdk/src/main/scala/com/horizen/certificatesubmitter/strategies/CeasingSidechain.scala

```diff
@@ -0,0 +1,78 @@
+package com.horizen.certificatesubmitter.strategies
+import com.horizen.certificatesubmitter.AbstractCertificateSubmitter.{ObsoleteWithdrawalEpochException, SignaturesStatus}
+import com.horizen.params.NetworkParams
+import com.horizen.utils.{BytesUtils, WithdrawalEpochInfo, WithdrawalEpochUtils}
+import com.horizen.websocket.client.{MainchainNodeChannel, WebsocketErrorResponseException, WebsocketInvalidErrorMessageException}
+import com.horizen.{AbstractHistory, AbstractState}
+import scorex.util.{ModifierId, ScorexLogging}
+import sparkz.core.NodeViewHolder.CurrentView
+
+import scala.util.{Failure, Success, Try}
+
+class CeasingSidechain(mainchainChannel: MainchainNodeChannel, params: NetworkParams)
```

Comment:
As an option, we may make this class parameterized by History and State, rather that have only the generic method inside.
I think it will feet better the whole design.

Reply:
discussed below